src/server/game/Spells/SpellInfo.cpp

```diff
@@ -3442,6 +3442,11 @@ bool _isPositiveEffectImpl(SpellInfo const* spellInfo, uint8 effIndex, std::unor
             // Starfall
             if (spellInfo->SpellFamilyFlags[2] == 0x00000100)
                 return false;
+            break;
+        case SPELLFAMILY_DEATHKNIGHT:
+            if (spellInfo->SpellIconID == 1933) // Ebon Plague
```

Comment:
Wasn't the idea to identify spells over the icon dropped? @Shauren

Reply:
Can change to SpellFamilyFlags, if needed

Reply:
it definitely was dropped - icons can change over patches (but ebon plague doesnt exist in legion)

Reply:
I know, but PR only just for 3.3.5. Maybe partialy can be useful for master

Reply:
@lost-illusion: It seems to me that Shauren is saying that the solution is fine since it does not affect Legion

Reply:
This solution can't affect legion because ebon plague doesnt exist in legion, but, im understand, i change iconid check to spell family